### DISCUSSION TITLE
Is there any way to use autocomplete on mac?

### DISCUSSION THREAD

**Discussion by gwsbhqt** [JanDeDobbeleer/oh-my-posh#3374] [Troubleshoot] [answered]
I know that autocomplete is not a function of ohmyposh. I can use psreadline on windows terminal, and zsh-autocomplete if i use ohmyzsh on mac. So if I use ohmyposh, do I need to mix it with ohmyzsh? Is there any way to use autocomplete on mac?

**Comment by JanDeDobbeleer** [comment, marked as answer]
@gwsbhqt there are indeed a few options. The easiest is to use ohmyzsh, things are perfectly compatible. The other option is to source the completion scripts directly or use another plugin manager. This [repo](https://github.com/zsh-users/zsh-completions) has some good options listed, might even be what you're looking for.

**Reply by gwsbhqt** [reply]
I don't quite understand what you mean. In my opinion, ohmyposh and ohmyzsh are competing products and mutual exclusion, isn't that the case?
Do you mean, ohmyposh and ohmyzsh are used in combination? Ohmyposh is responsible for the theme, and ohmyzsh is responsible for similar autocomplete plugins?
Or just replace ohmyposh with ohmyzsh?
Forgive my confusion and thank you for your answer.

**Reply by JanDeDobbeleer** [reply]
> Ohmyposh is responsible for the theme, and ohmyzsh is responsible for similar autocomplete plugins

@gwsbhqt exactly this. They are perfectly compatible and thus combine-able. Prompts is only one part of omz. Just make sure to source omz before oh-my-posh.
